### PR TITLE
feat: impl literal only PromQL query

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- refactor `EmptyMetrics` plan to accept an expr as data source
- implement literal expression, and literal-only binary expression

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

part of #1042

### CODE REVIEW

**Comment by waynexia** [comment]
A known format issue is tracked in https://github.com/GreptimeTeam/greptimedb/issues/1642

File: src/promql/src/extension_plan/empty_metric.rs [hunk 1]
```diff
@@ -242,8 +288,34 @@ impl Stream for EmptyMetricStream {
...
 
+/// Build a schema that only contains **millisecond** timestamp column
+fn build_ts_only_schema(column_name: &str) -> DFSchema {
+    let ts_field = DFField::new(
+        Some(""),
+        column_name,
+        DataType::Timestamp(TimeUnit::Millisecond, None),
+        false,
+    );
+    // safety: should not fail (UT covers this)
```
Comment (etolbakov) [inline src/promql/src/extension_plan/empty_metric.rs:299]: @waynexia just for my understanding what does `UT` stand for?

**Review by etolbakov — APPROVED** [review]
LGTM

File: src/promql/src/extension_plan/empty_metric.rs (hunk 1 above)
Comment (killme2008) [reply src/promql/src/extension_plan/empty_metric.rs:299]: Looks like it means unit test here.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**